src/node_api.cc

```diff
@@ -1024,6 +1024,27 @@ napi_status napi_delete_property(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+NAPI_EXTERN napi_status napi_has_own_property(napi_env env,
+                                              napi_value object,
+                                              napi_value key,
+                                              bool* result) {
+  NAPI_PREAMBLE(env);
+  CHECK_ARG(env, key);
+
+  v8::Isolate* isolate = env->isolate;
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  v8::Local<v8::Object> obj;
+
+  CHECK_TO_OBJECT(env, context, obj, object);
+  v8::Local<v8::Value> k = v8impl::V8LocalValueFromJsValue(key);
+  RETURN_STATUS_IF_FALSE(env, k->IsName(), napi_name_expected);
```

Comment:
You're right, I was looking at [a comment for has()](https://github.com/v8/v8/blame/master/include/v8.h#L3138).  Sorry for the confusion.  I think we should make sure though that it's obvious, that the n-api function is not identical to JS behavior, in particular that we'll never use the `ToPrimitive` symbol if the key is an object. (We should probably point that out in the V8 API, but there it might be less of an issue because we only allow `names`.)